packages/backend/src/models/NotificationsModel/NotificationsModel.ts

```diff
@@ -0,0 +1,125 @@
+import { Notification } from '@lightdash/common';
+import { Knex } from 'knex';
+import {
+    DashboardTileCommentsTableName,
+    DbDashboardTileComments,
+} from '../../database/entities/comments';
+import { DashboardsTableName } from '../../database/entities/dashboards';
+import {
+    DbNotifications,
+    NotificationsTableName,
+} from '../../database/entities/notifications';
+import { UserTableName } from '../../database/entities/users';
+
+type NotificationsModelDependencies = {
+    database: Knex;
+};
+
+export class NotificationsModel {
+    private readonly database: Knex;
+
+    constructor(deps: NotificationsModelDependencies) {
+        this.database = deps.database;
+    }
+
+    async getDashboardCommentNotifications(
+        userUuid: string,
+    ): Promise<Notification[]> {
+        const notificationsWithUserAndDashboardAndAuthor: (DbNotifications &
+            Pick<DbDashboardTileComments, 'dashboard_tile_uuid'> & {
+                user_first_name: string;
+                user_last_name: string;
+                dashboard_name: string;
+                author_first_name: string;
+                author_last_name: string;
+            })[] = await this.database(NotificationsTableName)
+            .join(
+                UserTableName,
+                `${NotificationsTableName}.user_uuid`,
+                '=',
+                `${UserTableName}.user_uuid`,
+            )
+            .join(
+                DashboardTileCommentsTableName,
+                `${NotificationsTableName}.comment_id`,
+                '=',
+                `${DashboardTileCommentsTableName}.comment_id`,
+            )
+            .join(
+                `${UserTableName} as comment_authors`,
+                `${DashboardTileCommentsTableName}.user_uuid`,
+                '=',
+                'comment_authors.user_uuid',
+            )
+            .join(
+                DashboardsTableName,
+                `${NotificationsTableName}.dashboard_uuid`,
+                '=',
+                `${DashboardsTableName}.dashboard_uuid`,
+            )
```

Comment:
Any ideas on optimising this query with these many joins? Or not too much of a concern at the moment...?

Reply:
done here: https://github.com/lightdash/lightdash/pull/9131/commits/7315bf4c9ca5f00ea3dd48ca996f0d3e7e3059a5 in `notificationsModel`